src/init.cpp

```diff
@@ -1859,17 +1856,10 @@ bool AppInitMain(const CoreContext& context, NodeContext& node, interfaces::Bloc
 
     // ********************************************************* Step 7b: Setup CoinJoin
 
-    ::coinJoinServer = std::make_unique<CCoinJoinServer>(*node.mempool, *node.connman, *::masternodeSync);
+    node.cj_ctx = std::make_unique<CJContext>(*node.connman, *node.mempool, *::masternodeSync, !ignores_incoming_txs);
 
 #ifdef ENABLE_WALLET
-    assert(!::coinJoinClientManagers);
-    ::coinJoinClientManagers = std::make_unique<CJClientManager>(*node.connman, *node.mempool, *::masternodeSync);
-
-    if (!ignores_incoming_txs) {
-        ::coinJoinClientQueueManager = std::make_unique<CCoinJoinClientQueueManager>(*node.connman, *::masternodeSync);
-    }
-
-    g_wallet_init_interface.InitCoinJoinSettings(*::coinJoinClientManagers);
+    g_wallet_init_interface.InitCoinJoinSettings(*node.cj_ctx->clientman);
```

Comment:
I think better to call `g_wallet_init_interface.InitCoinJoinSettings()` inside CJContext's constructor to hide implementation inside.

Reply:
IMO registering subsystems with interfaces should be considered a part of the program-wide application initialization process rather than as part of a subsystem that, unless working with it directly, the reader should not concern themselves with. I think it's fine as-is.